### PR TITLE
feat: identify teams & their billing in Posthog

### PR DESCRIPTION
**Pull request by ShubhamPalriwala**
## What does this PR do?
Groups users also by their Teams & sets their name & billing plan there! This is in addition to our current environment grouping!

## How should this be tested?

- Test A
- Test B

### Required

- [ ] Filled out the "How to test" section in this PR
- [ ] Read [How we Code at Formbricks](<[https://github.com/formbricks/formbricks/blob/main/CONTRIBUTING.md](https://formbricks.com/docs/contributing/how-we-code)>)
- [ ] Self-reviewed my own code
- [ ] Commented on my code in hard-to-understand bits
- [ ] Ran `pnpm build`
- [ ] Checked for warnings, there are none
- [ ] Removed all `console.logs`
- [ ] Merged the latest changes from main onto my branch with `git pull origin main`
- [ ] My changes don't cause any responsiveness issues

### Appreciated

- [ ] If a UI change was made: Added a screen recording or screenshots to this PR
- [ ] Updated the Formbricks Docs if changes were necessary

### CODE REVIEW

File: apps/web/app/(app)/environments/[environmentId]/components/PosthogIdentify.tsx [hunk 1]
```diff
@@ -5,24 +5,28 @@ import { usePostHog } from "posthog-js/react";
...
   environmentId: string;
+  team: TTeam;
 }) {
   const posthog = usePostHog();
 
   useEffect(() => {
     if (posthogEnabled && session.user && posthog) {
       posthog.identify(session.user.id, { name: session.user.name, email: session.user.email });
       posthog.group("environment", environmentId, { name: environmentId });
+      posthog.group("team", team.id, { name: team.name, billing: team.billing });
```
Comment (mattinannt) [inline apps/web/app/(app)/environments/[environmentId]/components/PosthogIdentify.tsx:27]: `team.billing` is a json object, right?
I would prefer that we give single string or boolean values to posthog to be able to filter by them.
e.g.
```
 posthog.group("team", team.id, { name: team.name, inAppSurveyBillingStatus, linkSurveyBillingStatus, userTargetingBillingStatus});
```
with the status `z.enum(["active", "cancelled", "inactive"]).default("inactive"),`

File: apps/web/app/(app)/environments/[environmentId]/components/PosthogIdentify.tsx [hunk 2]
```diff
@@ -5,24 +5,28 @@ import { usePostHog } from "posthog-js/react";
...
 const posthogEnabled = env.NEXT_PUBLIC_POSTHOG_API_KEY && env.NEXT_PUBLIC_POSTHOG_API_HOST;
 
 export default function PosthogIdentify({
   session,
   environmentId,
+  team,
 }: {
   session: Session;
   environmentId: string;
+  team: TTeam;
```
Comment (mattinannt) [inline apps/web/app/(app)/environments/[environmentId]/components/PosthogIdentify.tsx:19]: Ideally we would also pass `inAppSurveyBillingStatus`, `linkSurveyBillingStatus`, `userTargetingBillingStatus` here to not pass too many data to the client and save bandwidth.

**Review by mattinannt — COMMENTED** [review]
@ShubhamPalriwala thanks for the PR 😊💪
A few suggestions:

**Comment by ShubhamPalriwala** [comment]
AYe, nice idea!
Donezo! This is how the Team properties look now:
image

**Review by mattinannt — APPROVED** [review]
@ShubhamPalriwala thank you for making the changes; looks good 😊🚀💪

**Pull request merged**